src/components/credit-card-number/code/views/creditCardNumberView.jsx

```diff
@@ -0,0 +1,105 @@
+var React = require('react');
+
+module.exports = React.createClass({
+
+  propTypes: {
+    accepted: React.PropTypes.array,
+    cardNumber: React.PropTypes.string,
+    cardType: React.PropTypes.oneOf(['amex', 'discover', 'mastercard', 'visa']),
+    defaultValue: React.PropTypes.string,
+    id: React.PropTypes.string,
+    label: React.PropTypes.string,
+    name: React.PropTypes.string
+  },
+
+  getInitialState: function() {
+
+    var card = this.updateCard(this.props.defaultValue);
+
+    return {
+      cardNumber: card.cardNumber || '',
+      cardType: card.cardType || ''
+    };
+  },
+
+  getDefaultProps: function() {
+    return {
+      accepted: ['amex', 'discover', 'mastercard', 'visa'],
+      formatInput: false,
+      id: 'credit-card-number',
+      name: 'credit-card-number',
+      placeHolder: '• • • •   • • • •   • • • •   • • • •',
+      showImages: false
+    };
+  },
+
+  updateCard: function(cardNumber) {
+
+    if(typeof cardNumber === 'undefined'){
+      return false;
+    }
+
+    var cardInfo = this.getCard(cardNumber);
+    var formattedCard = (this.props.formatInput) ? cardNumber.replace(/(.{4})/g, '$1 ').trim() : cardNumber;
+
+    var card = {
+      cardNumber: formattedCard,
+      cardType: cardInfo.type
+    };
+
+    this.setState(card);
+
+    return card;
+  },
+
+  getCard: function(cardNumber) {
```

Comment:
May be overkill, but there's a way to get card type in a handy little library: https://github.com/Shortbreaks/iinChecker